Add release notes for 3.5.0 (#13051)
docker/docker.github.io: docker-for-windows/release-notes.md

### Bug fixes and minor changes

- **Volume Management**
  - Able to remove a file or directory from the Docker Dashboard.
  - Show more last time modified and size of the contents in the volumes.
  - The files inside the volume can be saved form the Docker Dashboard.
- Fixed an issue that caused credStore timeout errors when running the `docker login` command. Fixes [docker/for-win#11472](https://github.com/docker/for-win/issues/11472)
- Docker Desktop now allows the WSL 2 integration agent to start even when `/etc/wsl.conf` is malformed.
- Fixed an issue with the Docker Compose app not being stopped or removed when started by multiple configuration files. [docker/for-win#11445](https://github.com/docker/for-win/issues/11445)
- Fixed a bug where Docker Desktop fails to restart after a power failure because the Hyper-V VM restarted prematurely.
- Fixed a link to the policy that provides details on how Docker handles the uploaded diagnostics data. Fixes [docker/for-mac#5741](https://github.com/docker/for-mac/issues/5741)

> - The default Windows named pipe name is now `\\.\pipe\dockerDesktopWindowsEngine` and the Linux named pipe name is `\\.\pipe\dockerDesktopLinuxEngine`. For backwards compatibility `\\.pipe\docker_engine` is still available. We recommend using the `docker context`s called `desktop-linux` and `desktop-windows`.

Reviewer: Similarly to the Mac case above, perhaps this one change should become:

- The default `docker` CLI `context` is now `desktop-linux` in Linux containers mode and `desktop-windows` when in Windows containers mode.
- The Linux engine Docker API is now available on `\\.\pipe\dockerDesktopLinuxEngine`.
- The Windows engine Docker API is now available on `\\.\pipe\dockerDesktopWindowsEngine`.

Reviewer 2: Is this a thing we want to advertise to the user though? Shouldn't it be a black box?

[Download](https://desktop.docker.com/win/stable/amd64/Docker%20Desktop%20Installer.exe?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-win-amd64){: .button .primary-btn}

> ### New

Reviewer 3: nit: add space after the heading

- Dev environments

> ### Upgrades

Reviewer 3: nit: add space before and after the heading

- [Compose V2 beta](https://github.com/docker/compose-cli/releases/tag/v2.0.0-beta.4)
- [Kubernetes 1.21.2](https://github.com/kubernetes/kubernetes/releases/tag/v1.21.2)

### Bug fixes and minor changes

> - Fixed an issue that caused credStore timeout errors when running docker login. Fixes [docker/for-win#11472](https://github.com/docker/for-win/issues/11472)

Reviewer 3: ```suggestion
- Fixed an issue that caused credStore timeout errors when running the `docker login` command. Fixes [docker/for-win#11472](https://github.com/docker/for-win/issues/11472)
```

- Fixed a bug where a container cannot be started when a file is bind-mounted into a nested mountpoint. Fixes [docker/compose-cli#1795](https://github.com/docker/compose-cli/issues/1795).

> - Allow the WSL 2 integration agent to start even if `/etc/wsl.conf` is malformed.

Reviewer 3: ```suggestion
- Docker Desktop now allows the WSL 2 integration agent to start even when `/etc/wsl.conf` is malformed.
```

> - Fixed an issue with docker compose app not being stopped or removed when started by multiple configuration files [docker/for-win#11445](https://github.com/docker/for-win/issues/11445)

Reviewer 3: ```suggestion
- Fixed an issue with the Docker Compose app not being stopped or removed when started by multiple configuration files. [docker/for-win#11445](https://github.com/docker/for-win/issues/11445)
```

- Fixed a bug where Docker Desktop fails to restart after a power failure because the Hyper-V VM restarted prematurely.

> - Fix link to policy regarding uploaded diagnostics data. Fixes [docker/for-mac#5741](https://github.com/docker/for-mac/issues/5741)

Reviewer 3: See comment in the Mac section

## Docker Desktop 3.3.2
2021-05-03

> > [Download](https://desktop.docker.com/win/stable/amd64/63878/Docker%20Desktop%20Installer.exe?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-win-amd64)

Reviewer 4: The UTM is not needed here.

Author: arg, missed this change. I'll update it in a bit
